up-l1/iceoryx2.adoc

```diff
@@ -0,0 +1,264 @@
+--
+All types of uProtocol messages *MUST* be transferred using iceoryx2's PublishSubscribe MessagingPattern only. iceoryx2's RequestResponse API *MUST NOT* be used.
+--
+
+== UMessage Mapping
+
+A uProtocol message consists of _UAttributes_ and optional payload. The following sections define how these are mapped to/from iceoryx2 Services.
+
+=== UAttributes
+
+[.specitem,oft-sid="dsn~up-transport-iceoryx2-attributes-mapping~1",oft-needs="impl,utest",oft-tags="TransportLayerImpl"]
+--
+The value of a iceoryx2 Service that is used to convey a uProtocol message *MUST* have a https://docs.rs/iceoryx2/0.6.1/iceoryx2/service/builder/publish_subscribe/struct.Builder.html#method.user_header[user header] as defined in <<uAttributes Mapping to iceoryx2 user header>>.
+--
+
+==== uAttributes Mapping to iceoryx2 user header
+
+[source,rust]
+----
+const MAX_FEASIBLE_UATTRIBUTES_SERIALIZED_LENGTH: usize = 1000; // choosing ~1000 u8s
+                                                                // somewhat arbitrarily
+                                                                // this should be confirmed
+
+#[repr(C)]
+#[derive(ZeroCopySend)]
+struct UProtocolHeader {
+    uprotocol_schema_version: u8,
+    uattributes_serialized: FixedSizeVec<u8, MAX_FEASIBLE_UATTRIBUTES_SERIALIZED_LENGTH>
+}
+----
+
+.uAttributes Mapping to UProtocolHeader, iceoryx2 User Header
+[%autowidth]
+|===
+| field | value
+
+| `uprotocol_schema_version`
+a| *MUST* be set to a single byte representing the UProtocol major version (`0x01` at the time of writing)
+
+| `uattributes_serialized`
+a| *MUST* be set to the bytes representing the protobuf encoding of the UAttributes object as defined by the link:../up-core-api/uprotocol/v1/uattributes.proto[UAttributes Protocol Buffers definition]
+
+|===
+
+_Known limitations_: Using a `FixedSizeVec` in the header containing a serialization of the entire `UAttributes` is wasteful and unnecessarily makes every message larger than it really need be most of the time. The approach chosen here was chosen primarily for expediency and simplicity of implementation. It should be possible to optimize this significantly. Some low-hanging fruit: use custom structs for entirety of `UAttributes` in User Header, make use of `Option<T>` to cut down on size when e.g. not using token or traceparent.
+
+==== Message Priority
+
+Currently not a clear path to map from uProtocol message priority to iceoryx2 QoS (Quality of Service). Skip for current revision.
+
+=== Payload Mapping
+
+[.specitem,oft-sid="dsn~up-transport-iceoryx2-payload-mapping~1",oft-needs="impl,utest",oft-tags="TransportLayerImpl"]
+--
+An iceoryx2 message that is used to convey a uProtocol message *MUST* contain in its payload the unaltered value of the UMessage's _payload_ field.
+--
+
+[.specitem,oft-sid="dsn~up-transport-iceoryx2-payload-mapping~2",oft-needs="impl,utest",oft-tags="TransportLayerImpl"]
```

Comment:
```suggestion
[.specitem,oft-sid="dsn~up-transport-iceoryx2-payload-zero-copy~1",oft-needs="impl,utest",oft-tags="TransportLayerImpl"]
```

this needs to get a different name, the number at the end of the ID is the *revision* of the specitem ...